crates/types/src/scope.rs

```diff
@@ -0,0 +1,60 @@
+use core::fmt;
+
+use bitcoin::{
+    consensus::{encode::Error, Decodable, Encodable},
+    hashes::{sha256, Hash},
+    io::Cursor,
+};
+use serde::{Deserialize, Serialize};
+
+#[derive(Clone, Copy, PartialEq, Eq, PartialOrd, Ord, Debug, Deserialize, Serialize)]
+pub struct Scope(sha256::Hash);
```

Comment:
This deserves a docstring.

Reply:
I'm not sure what to write here, as what is hashed specifically can be changed depending on operator implementation. Will something like this be okay?

```suggestion
/// A unique identifier of deposit setup made by hashing unique data related to it
#[derive(Clone, Copy, PartialEq, Eq, PartialOrd, Ord, Debug, Deserialize, Serialize)]
pub struct Scope(sha256::Hash);
```